### PR TITLE
Add trigger capability on github pr review

### PR DESCRIPTION
**Pull request by omgjlk**
A github pull request review is a new type of event that happens when
user provides a review of a pull request
https://developer.github.com/v3/pulls/reviews

It will create an event
https://developer.github.com/v3/activity/events/types/#pullrequestreviewevent
sent to the webhook. Initially allow for filtering on the review state,
which can be one of three choices.

Change-Id: If94c2fe8adbe18fa7f426d5462559ba24963424c
(cherry picked from commit b68be427451fcf700cb2a65501b027272d75cd01)

Closes BonnyCI/projman#73

### CODE REVIEW

File: zuul/connection/github.py [hunk 1]
```diff
@@ -165,6 +165,26 @@ def _event_issue_comment(self, request):
...
+        if action != 'submitted':
+            return
+        pr_body = body.get('pull_request')
+        if pr_body is None:
+            return
+
+        review = body.get('review')
+        if review is None:
+            return
+
```
Comment (rattboi) [inline zuul/connection/github.py:181]: What's the reasoning here that review exist, if that value is never used? Can you get a pr review event with an empty review?

File: tests/test_github.py [hunk 2]
```diff
@@ -100,6 +100,18 @@ def test_comment_unmatched_event(self):
         self.waitUntilSettled()
         self.assertEqual(0, len(self.history))
 
+    def test_review_event(self):
+        A = self.fake_github.openFakePullRequest('org/project', 'master', 'A')
+        self.fake_github.emitEvent(A.getReviewAddedEvent('approved'))
+        self.waitUntilSettled()
+        self.assertEqual(3, len(self.history))
```
Comment (rattboi) [inline tests/test_github.py]: This looks good, but it's not obvious to me what the assert is testing (I see it looks similar to the other tests, but why history == 3 vs history ==1?)

**Review by rattboi — COMMENTED** [review]
I don't have any changes to request, it looks ok to me, if you could clarify the few things I mentioned.

File: zuul/connection/github.py (hunk 1 above)
Comment (omgjlk) [reply zuul/connection/github.py:181]: I'm.. not really sure. Honestly I copied it from another event handler.

File: tests/test_github.py (hunk 2 above)
Comment (omgjlk) [reply tests/test_github.py]: that's a good question. Basically I copied it from the other tests. I think the history is the number of things in the pipeline perhaps?

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/check_github/BonnyCI/zuul/7/1484262315.15/tox_py27 : FAILURE in 2h 05m 10s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/check_github/BonnyCI/zuul/7/1484341763.22/tox_py27 : FAILURE in 14m 56s

File: zuul/connection/github.py [hunk 3]
```diff
@@ -165,6 +165,26 @@ def _event_issue_comment(self, request):
...
+        event = self._pull_request_to_event(pr_body)
+        event.state = body.get('review').get('state')
```
Comment (gandelman-a) [inline zuul/connection/github.py]: line 178 already get()s the review, could avoid the extra dict lookup here by referencing 'review'

File: tests/test_github.py [hunk 4]
```diff
@@ -501,4 +515,4 @@ def test_failed_change_at_head(self):
         self.assertTrue(C.is_merged)
         self.assertNotIn('merge', A.labels)
         self.assertNotIn('merge', B.labels)
-        self.assertNotIn('merge', C.labels)
\ No newline at end of file
+        self.assertNotIn('merge', C.labels)
```
Comment (gandelman-a) [inline tests/test_github.py:518]: this looks like unnecessary cruft

**Review by gandelman-a — COMMENTED** [review]
just a couple of very small nits. otherwise this is pretty straight forward and LGTM

File: zuul/connection/github.py (hunk 3 above)
Comment (omgjlk) [reply zuul/connection/github.py]: Fixing.

File: tests/test_github.py (hunk 4 above)
Comment (omgjlk) [reply tests/test_github.py:518]: Vim is doing that. Whoever edited this file before wasn't using a proper editor. :)

**Review by gandelman-a — APPROVED** [review]

File: zuul/connection/github.py (hunk 1 above)
Comment (ConsoleCatzirl) [reply zuul/connection/github.py:181]: This makes sense to me... we're making sure that the body object has a review key before transforming the entire body object into an event.

**Review by ConsoleCatzirl — APPROVED** [review]

**Comment by anne-bonny** [comment]
Starting gate_github jobs.

**Pull request merged**